### PR TITLE
tools/render-template: Allow disabling yaml loading

### LINKED ISSUE #4487: tmpl files not rendered by `render-cloudcfg`
Our Makefile has some rules that depend on `render-cloudcfg` being able to render generic `.tmpl` files. In this particular case, it is trying to render `systemd/cloud-init-generator.tmpl`:
```
render-template:
	$(PYTHON) ./tools/render-cloudcfg --variant=$(VARIANT) $(FILE) $(subst .tmpl,,$(FILE))
...
benchmark-generator: FILE=$(GENERATOR_F).tmpl
benchmark-generator: export ITER=$(NUM_ITER)
benchmark-generator: render-template
        $(BENCHMARK) $(GENERATOR_F)
```

However, the `render-cloudcfg` script eventually [attempts to load yaml](https://github.com/canonical/cloud-init/blob/main/cloudinit/templater.py/#L168). This works fine for rendered cloud-config, but it breaks for generic jinja templated scripts like `cloud-init-generator.tmpl`. 

The actual rendering code used on package build is [in setup.py](https://github.com/canonical/cloud-init/blob/main/setup.py/#L53). It looks like it may just be ignoring the error since the rendering portion works before hitting the error, and the script is exiting 0. We should create a new `render-template` script that can be used generically and fix the load yaml error.

### PR DESCRIPTION
**Pull request by holmanb**
## Proposed Commit Message

```
tools/render-template: Allow disabling yaml loading

Fixes GH-4487
```
Fixes #4487

### CODE REVIEW

**Review by igalic — APPROVED** [review]
that seems most sensible!

**Review by blackboxsw — CHANGES_REQUESTED** [review]
I `git bisected` this error/behavior down to commit 0044c83512d915137ccc79553b57c0250b0f474f which actually fixed a broken global logging setup relative to tools/render-cloudcfg globally which now allowed render warning logs to `stderr` which had been there during package build for quite a while.

Given that the initial intent of the yaml_load was to validate that the cloud.cfg we packaged was valid YAML, I think it's worth trying to treat this as a harder error so we downstreams can catch it at package build time and prevent shipping an invalid YAML in `/etc/cloud/cloud.cfg`.   


So, what I think needs to happen here is setup.py should be aware of which files are expected to by YAML and which aren't when rendering from templates. In practice it looks like it's just `/etc/cloud/cloud.cfg`.

In order to allow setup.py to continue to validate the YAML of `/tec/cloud/cloud.cfg` should we grow an option to `setup.render_tmpl` like the following?

```diff
diff --git a/setup.py b/setup.py
index 5fb269655..4344a8312 100644
--- a/setup.py
+++ b/setup.py
@@ -50,7 +50,7 @@ VARIANT = None
 PREFIX = None
 
 
-def render_tmpl(template, mode=None):
+def render_tmpl(template, mode=None, is_yaml=False):
     """render template into a tmpdir under same dir as setup.py
 
     This is rendered to a temporary directory under the top level
@@ -82,6 +82,7 @@ def render_tmpl(template, mode=None):
         cmd_variant = ["--variant", VARIANT]
     if PREFIX:
         cmd_prefix = ["--prefix", PREFIX]
+    cmd_yaml = [] if is_yaml else ["--no-yaml"]
     subprocess.run(
         [
             sys.executable,
@@ -269,7 +270,7 @@ if not in_virtualenv():
         INITSYS_ROOTS[k] = "/" + INITSYS_ROOTS[k]
 
 data_files = [
-    (ETC + "/cloud", [render_tmpl("config/cloud.cfg.tmpl")]),
+    (ETC + "/cloud", [render_tmpl("config/cloud.cfg.tmpl", is_yaml=True)]),
     (ETC + "/cloud/clean.d", glob("config/clean.d/*")),
     (ETC + "/cloud/cloud.cfg.d", glob("config/cloud.cfg.d/*")),
     (ETC + "/cloud/templates", glob("templates/*")),
```

File: cloudinit/templater.py [hunk 1]
```diff
@@ -161,11 +161,16 @@ def render_string(content, params):
 
 
 def render_cloudcfg(variant, template, output, prefix=None):
+    render_template(variant, template, output, False, prefix)
```
[the remark this answers is not in the document]
Comment (holmanb) [reply cloudinit/templater.py:164]: originally created this for compat - realize now that this function is only called in tests so I'll just strike this extra signature

**Comment by holmanb** [comment]
> I git bisected this error/behavior down to commit https://github.com/canonical/cloud-init/commit/0044c83512d915137ccc79553b57c0250b0f474f which actually fixed a broken global logging setup relative to tools/render-cloudcfg globally which now allowed render warning logs to stderr which had been there during package build for quite a while.

I was curious why that happened. Fixing one bug exposed another. Nice.

> Here's a diff that'd only build INITSYS_FILES for the init-system provided.

Funky use for lambdas, I like it.

@blackboxsw I think I've addressed everything, thanks for the review.

**Review by blackboxsw — APPROVED** [review]
LGTM thanks for the fixup.

**Pull request merged**